src/app/space/create/deployments/apps/delete-deployment-modal.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, ElementRef, EventEmitter, Input, OnInit, Output, ViewChild, ViewEncapsulation } from '@angular/core';
+
+import { NotificationType } from 'ngx-base';
+import { ModalDirective } from 'ngx-bootstrap/modal';
+import { Subscription } from 'rxjs';
+import { Environment } from '../models/environment';
+
+import { DeploymentsService } from '../services/deployments.service';
+import { DeploymentCardComponent } from './deployment-card.component';
+
+@Component({
+  host: {
+    class: 'create-dialog'
+  },
+  encapsulation: ViewEncapsulation.None,
+  selector: 'delete-deployment-modal',
+  templateUrl: './delete-deployment-modal.component.html'
+})
+export class DeleteDeploymentModal {
+
+  @Input() host: ModalDirective;
+  @Output() deleteEvent = new EventEmitter();
+
+  applicationId: string = '';
+  environmentName: string = '';
+
+  constructor(
+    private deploymentsService: DeploymentsService) {
+  }
+
+  public openModal(applicationId: string, environmentName: string): void {
+    this.applicationId = applicationId;
+    this.environmentName = environmentName;
```

Comment:
Could `applicationId` and `environmentName` be converted to `@Input` properties of the modal, rather than being passed in as params to this function call?

Reply:
Good call.